rust/toolchain.bzl

```diff
@@ -151,22 +174,29 @@ def _make_libstd_and_allocator_ccinfo(ctx, rust_lib, allocator_library):
             order = "topological",
         )
 
-        link_inputs.append(cc_common.create_linker_input(
+        link_inputs = cc_common.create_linker_input(
             owner = rust_lib.label,
             libraries = std_inputs,
-        ))
+        )
 
-    allocator_inputs = None
-    if allocator_library:
-        allocator_inputs = [allocator_library[CcInfo].linking_context.linker_inputs]
+        allocator_inputs = None
```

Comment:
Why this change? In general allocator_library is independent from the `rust_stdlib_info`. I can imagine a scenario where people want to use the system installed rust stdlib (therefore without declaring inputs in the `rust_toolchain`, but they still want to add `allocator_library` explicitly.

In general `allocator_library` contains symbols that normally rustc generates on demand when producing a binary. When using clang/lld obviously they don't know how to produce those symbols, so the linking fails. Allocator_library is responsible for defining those symbols and this library is typically part of the toolchain (it currently cannot be system installed because clang needs to be told should use it).

Reply:
(the change I'm talking about is indenting the code to the right, so it is only executed when there are `std_rlibs`)

Reply:
The code today only returns a `CcInfo` if `link_inputs` has been populated which only happens when `std_rlibs` is populated.:
https://github.com/bazelbuild/rules_rust/blob/8cca226c6bc3c55444c98a6a08e7d19b6dd028f2/rust/toolchain.bzl#L159-L170

This is not a change in behavior and I thought it'd be clearer to gate this behind the same check. If this is incorrect then I'd say it was originally a bug that I'm happy to do something about but I otherwise think this is a good improvement that makes the new approach more readable.

Reply:
You're right. Go ahead :)